File: dashboard/src/store/market.ts
import { create } from 'zustand'
import { api } from '../api/client'

interface AccountData {
  connected: boolean
  balance: number
  equity: number
  margin: number
  free_margin: number
  profit: number
}

interface MarketState {
  account: AccountData | null
  ticks: Record<string, { bid: number; ask: number; timestamp: string }>
  settings: any
  fetchAccount: () => Promise<void>
  fetchSettings: () => Promise<void>
  updateTick: (symbol: string, bid: number, ask: number, timestamp: string) => void
}

export const useMarketStore = create<MarketState>((set) => ({
  account: null,
  ticks: {},
  settings: null,

  fetchAccount: async () => {
    try {
      const account = await api.getAccount()
      set({ account })
    } catch { /* offline */ }
  },

  fetchSettings: async () => {
    try {
      const settings = await api.getSettings()
      set({ settings })
    } catch { /* offline */ }
  },

  updateTick: (symbol, bid, ask, timestamp) => {
    set((state) => ({
      ticks: { ...state.ticks, [symbol]: { bid, ask, timestamp } },
    }))
  },
}))

File: agent/api/market.py
"""Live market data endpoints."""

from fastapi import APIRouter, Depends

from agent.api.auth import get_current_user

router = APIRouter(prefix="/api", tags=["market"])


@router.get("/market/{symbol}")
async def get_market_data(symbol: str, user: str = Depends(get_current_user)):
    """Get current price and indicator snapshot for a symbol."""
    from agent.api.main import app_state
    bridge = app_state["bridge"]

    tick = await bridge.get_tick(symbol)
    if not tick:
        return {"symbol": symbol, "connected": False}

    return {
        "symbol": symbol,
        "connected": True,
        "bid": tick.bid,
        "ask": tick.ask,
        "spread": tick.spread,
        "timestamp": tick.timestamp.isoformat(),
    }


@router.get("/account")
async def get_account(user: str = Depends(get_current_user)):
    """Get MT5 account info."""
    from agent.api.main import app_state
    account = await app_state["bridge"].get_account()
    if not account:
        return {"connected": False}
    return {
        "connected": True,
        "balance": account.balance,
        "equity": account.equity,
        "margin": account.margin,
        "free_margin": account.free_margin,
        "margin_level": account.margin_level,
        "profit": account.profit,
    }
